src/x86_64/kernel_machine.h

```diff
@@ -239,15 +249,16 @@ static inline cpuinfo current_cpu(void)
     return (cpuinfo)pointer_from_u64(addr);
 }
 
+extern u8 use_xsave;
+
 static inline u64 extended_frame_size(void)
 {
-#if 0
-    u32 v[4];
-    cpuid(0xd, 0, v);
-    return v[1];
-#else
-    return 512;                 /* XXX fx only right now */
-#endif
+    if (use_xsave) {
+        u32 v[4];
+        cpuid(0xd, 0, v);
```

Comment:
I guess this won't be called too frequently - just when a new frame is allocated or clone(2) is called - but it might save some vm exits if the xsave size is probed once on init and stashed away.

Reply:
I realized that the kernel frames are allocated before the xsave or avx gets discovered, so I had to rework that part a bit to initialize the cr0/cr4/xcr0 settings on cpu 0 before entering the kernel runtime init so that those frames would be properly allocated, but now the size is only calculated once at init.